featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -500,6 +552,8 @@ def _build_transform_features(self, all_features, entity, max_depth=0,
                                                         input_types,
                                                         trans_prim,
                                                         require_direct_input=require_direct_input)
+            matching_inputs = filter_matches_by_options(matching_inputs,
```

Comment:
is it possible to move the call to `filter_matches_by_options` into the `self._get_matching_inputs`? I think that would be more logical since the caller of `_get_matching_inputs` would like to assume it is already only getting valid matches